components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -787,12 +802,22 @@ public boolean doPreSetUserClaimValues(String userName, Map<String, String> clai
         }
         IdentityUtil.clearIdentityErrorMsg();
         boolean accountLocked = Boolean.parseBoolean(claims.get(UserIdentityDataStore.ACCOUNT_LOCK));
+        boolean accountDisabled = Boolean.parseBoolean(claims.get(UserIdentityDataStore.ACCOUNT_DISABLED));
         if (accountLocked) {
             IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(UserCoreConstants
                     .ErrorCode.USER_IS_LOCKED);
             IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
         }
 
+        else if (accountDisabled) {
+            IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(
+                    IdentityCoreConstants.USER_ACCOUNT_DISABLED_ERROR_CODE);
+            IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
+        }
+        else{
```

Comment:
yes, it's not needed. Added as best pratice to avoid confusion for future developers, by clealry defining the margins of if else statement